motionplan/motionPlanner.go

```diff
@@ -381,3 +387,115 @@ IK:
 	}
 	return orderedSolutions, nil
 }
+
+// CheckPlan checks if obstacles intersect the trajectory of the frame following the plan.
+func CheckPlan(
+	checkFrame frame.Frame,
+	plan Plan,
+	worldState *frame.WorldState,
+	fs frame.FrameSystem,
+	currentPosition spatialmath.Pose,
+	currentInputs []frame.Input,
+	errorState spatialmath.Pose,
+	logger *zap.SugaredLogger,
+) error {
+	// ensure that we can actually perform the check
+	if len(plan) < 1 {
+		return errors.New("plan must have at least one element")
+	}
+	// construct solverFrame
+	// Note that this requires all frames which move as part of the plan, to have an
+	// entry in the very first plan waypoint
+	sf, err := newSolverFrame(fs, checkFrame.Name(), frame.World, plan[0])
+	if err != nil {
+		return err
+	}
+
+	// construct planager
+	sfPlanner, err := newPlanManager(sf, fs, logger, defaultRandomSeed)
+	if err != nil {
+		return err
+	}
+
+	// convert plan into nodes
+	planNodes, err := sf.planToNodes(plan)
+	if err != nil {
+		return err
+	}
+
+	// This should be done for any plan whose configurations are specified in relative terms rather than absolute ones.
+	// Currently this is only TP-space, so we check if the PTG length is >0.
+	// The solver frame will have had its PTGs filled in the newPlanManager() call, if applicable.
+	relative := len(sf.PTGs()) > 0
+
+	if relative {
+		// get pose of robot along the current trajectory it is executing
+		ptgs := sf.PTGs()
+		selectedPTG := ptgs[int(math.Round(currentInputs[0].Value))]
+		selectedTraj, err := selectedPTG.Trajectory(
+			currentInputs[1].Value, currentInputs[2].Value,
+		)
+		if err != nil {
+			return err
+		}
+		lastPose := spatialmath.NewZeroPose()
+		for _, trajNode := range selectedTraj {
+			if trajNode.Dist > currentInputs[2].Value {
+				break
+			}
+			lastPose = trajNode.Pose
+		}
```

Comment:
With some of the more recent updates, this shouldn't be necessary to do anymore.

Everything here can be replaced with: `lastPose := sf.Transform(currentInputs)`